File: csharp/secret-handshake/SecretHandshake.cs
using System;
using System.Collections.Generic;

public class SecretHandshake
{
  private static Dictionary<string, string> commands = new Dictionary<string, string> {
    { "1", "wink"},
    { "10", "double blink" },
    { "100", "close your eyes" },
    { "1000", "jump" }
  };

  public static string[] Commands (int x)
  {
    List<string> result = new List<string>();
    foreach (KeyValuePair<string, string> command in commands) {
      int key = Convert.ToInt32(command.Key, 2);
      if ((x & key) == key) result.Add(command.Value);
    }
    if (x > (1 << commands.Count)) result.Reverse();
    return result.ToArray();
  }
}

File: javascript/secret-handshake/secret-handshake.js
const SECRETS = [
  'wink',
  'double blink',
  'close your eyes',
  'jump'
];

const REVERSE = 4;

export const commands = (handshake) => {
  let secrets = SECRETS.filter((s, i) => (handshake & 1 << i) > 0);
  if ((handshake & 1 << REVERSE) > 0)
    secrets.reverse();
  return secrets;
};
